File: core/src/main/java/xavier/project_iodine/client/Window.java
package xavier.project_iodine.client;

import java.awt.Dimension;

import javax.swing.JFrame;

public class Window extends JFrame{
    public Window (int width, int height, String title, Game game) {
        super(title);
        game.setPreferredSize(new Dimension (width, height));
        game.setMaximumSize(new Dimension (width, height));
        game.setMinimumSize(new Dimension (width, height));
        addKeyListener(game);
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setResizable(false);
        setVisible(true);
        requestFocus();
        add(game);
        pack();
        setLocationRelativeTo(null);
    }
}

File: src/main/kotlin/client/renderer/Window.kt
package client.renderer

import java.awt.Dimension
import javax.swing.JFrame

class Window (name: String?, sizeX: Int, sizeY : Int) : JFrame(name) {
    init {
        val dim = Dimension(sizeX, sizeY)
        this.maximumSize = dim
        this.preferredSize = dim
        this.minimumSize = dim
        this.size = dim
    }
}